From 8e8b3e443a7f1d0644f43888c8683698f26d4d13 Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Sun, 25 Jul 2021 01:20:31 +0100
Subject: [PATCH 22/27] fstools: update to git HEAD

 bad1835 fstools: add partname volume driver
 19d7d93 libfstools: partname: several fixes
 3c38f0c libfstools: fix build with glibc
 d05ad93 libfstools: remove superflus include
 964d1e3 partname: allow skipping existing 'rootfs_data' partition
 c44b40b overlay: fix syncronizing typo
 b5397a1 fstools: block: fix segfault on mount with no target
 bd7cc8d block: use dynamically allocated target string
 6d8450e blockd: use allocated strings instead of fixed buffers
 d47909e libblkid-tiny: fix buffer overflow
 67d2297 block: match device path instead of assuming /dev/%s
 2aeba88 block: allow autofs and umount commands also on MTD/UBI
 3d40a1b blockd: add missing #define _GNU_SOURCE
 4d4dcfb blockd: detect mountpoint of /dev/mapper/*
 2f42515 block: resolve /dev/mapper/* name for /dev/dm-0 when hotplugging
 39558a1 blockd: also send ubus notification on mount hotplug
 3386b6b blockd: fix trigger name
 cdc9939 blockd: move to its own POSIX process group
 59f7c11 blockd: create mountpoint parent folder if needed
 9cc96af Revert "block: resolve /dev/mapper/* name for /dev/dm-0 when hotplugging"
 06334ac Revert "blockd: detect mountpoint of /dev/mapper/*"
 9ab3551 block: use /dev/dm-* instead of /dev/mapper/*
 5114595 block: allow remove hotplug event to arrive at blockd
 a846c6b blockd: fix length of timeout int passed to ioctl
 1d681ca block: support umount device basename

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
---
 package/system/fstools/Makefile          | 6 +++---
 package/system/fstools/files/blockd.init | 1 +
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/package/system/fstools/Makefile b/package/system/fstools/Makefile
index 2da508d541..b582a17bae 100644
--- a/package/system/fstools/Makefile
+++ b/package/system/fstools/Makefile
@@ -12,9 +12,9 @@ PKG_RELEASE:=1
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=$(PROJECT_GIT)/project/fstools.git
-PKG_MIRROR_HASH:=a485792d90c71cd4fb396ce97f42a57ee4d2a3d78e5f3fd0748270ffb14209e6
-PKG_SOURCE_DATE:=2021-01-04
-PKG_SOURCE_VERSION:=c53b18820756f6f32ad0782d3bf489422b7c4ad3
+PKG_MIRROR_HASH:=6a457b812166e04e2244ee1be92a4957666b5d1554315c0e18db1b30376cc617
+PKG_SOURCE_DATE:=2021-07-28
+PKG_SOURCE_VERSION:=cc63933faedd8d4fcdabb872cf4661ac04fe4ba2
 CMAKE_INSTALL:=1
 
 PKG_LICENSE:=GPL-2.0
diff --git a/package/system/fstools/files/blockd.init b/package/system/fstools/files/blockd.init
index a4ce57d40d..bdd8bbf622 100755
--- a/package/system/fstools/files/blockd.init
+++ b/package/system/fstools/files/blockd.init
@@ -16,6 +16,7 @@ reload_service() {
 start_service() {
 	procd_open_instance
 	procd_set_param command "$PROG"
+	procd_set_param watch block
 	procd_set_param respawn
 	procd_close_instance
 }
-- 
2.25.1

